Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenStick.cs

```diff
@@ -251,23 +262,63 @@ private void OnPointerDown(InputAction.CallbackContext ctx)
                 return;
 
             BeginInteraction(screenPosition, GetCameraFromCanvas());
+            if (touchControl != null)
+            {
+                m_TouchControl = touchControl;
+                m_PointerMoveAction.ApplyBindingOverride($"{touchControl.path}/position", path: "<Touchscreen>/touch*/position");
+            }
+
             m_PointerMoveAction.performed += OnPointerMove;
+            m_IsIsolationActive = true;
+        }
+
+        private void OnPointerChanged(InputAction.CallbackContext ctx)
+        {
+            if (ctx.control.IsPressed())
+                OnPointerDown(ctx);
+            else
+                OnPointerUp(ctx);
         }
 
         private void OnPointerMove(InputAction.CallbackContext ctx)
         {
             // only pointer devices are allowed
             Debug.Assert(ctx.control?.device is Pointer);
+            Vector2 screenPosition;
 
-            var screenPosition = ((Pointer)ctx.control.device).position.ReadValue();
+            // If it's a finger take the value from the finger that initiated the change
```

Comment:
More a general conceptual thought around the scalability of this than the particular fix, what if the on screen control is interacted with a pen?

Reply:
good question I don't have a pen/touch setup to test.
But we can fix the pen case after if needed. Here it only fix when it comes from a touch device